### ISSUE TITLE
Mock Backends of 5 qubits dosen't work for pulse.

### ISSUE THREAD

**Issue by UlysseCambier** [Qiskit/qiskit-aer#1113]
### Informations

-**Qiskit version**:0.23.3
- **Qiskit AER version**:0.7.3
- **Python version**:3.7.9
- **Operating system**:Microsoft Windows 10 Family, x64, 10.0.18362

### What is the current behavior?

Mock Backends that are said to support Open Pulse doesn't pass assemble() stage.
```
Traceback (most recent call last):
  File "C:/Users/cambi/AppData/Local/Programs/Python/Python37/testings.py", line 23, in <module>
    result = backend_mock.run(qobj).result()
  File "C:\Users\cambi\AppData\Local\Programs\Python\Python37\lib\site-packages\qiskit\providers\aer\aerjob.py", line 42, in _wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\cambi\AppData\Local\Programs\Python\Python37\lib\site-packages\qiskit\providers\aer\aerjob.py", line 101, in result
    return self._future.result(timeout=timeout)
  File "C:\Users\cambi\AppData\Local\Programs\Python\Python37\lib\concurrent\futures\_base.py", line 435, in result
    return self.__get_result()
  File "C:\Users\cambi\AppData\Local\Programs\Python\Python37\lib\concurrent\futures\_base.py", line 384, in __get_result
    raise self._exception
  File "C:\Users\cambi\AppData\Local\Programs\Python\Python37\lib\concurrent\futures\thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\cambi\AppData\Local\Programs\Python\Python37\lib\site-packages\qiskit\providers\aer\backends\aerbackend.py", line 251, in _run
    output = self._execute(qobj)
  File "C:\Users\cambi\AppData\Local\Programs\Python\Python37\lib\site-packages\qiskit\providers\aer\backends\pulse_simulator.py", line 254, in _execute
    return pulse_controller(qobj)
  File "C:\Users\cambi\AppData\Local\Programs\Python\Python37\lib\site-packages\qiskit\providers\aer\pulse\controllers\pulse_controller.py", line 143, in pulse_controller
    pulse_de_model.freqs = system_model.calculate_channel_frequencies(qubit_lo_freq=qubit_lo_freq)
  File "C:\Users\cambi\AppData\Local\Programs\Python\Python37\lib\site-packages\qiskit\providers\aer\pulse\system_models\pulse_system_model.py", line 205, in calculate_channel_frequencies
    qscale = u_lo_idx.scale.real
AttributeError: 'list' object has no attribute 'real'
```

### Steps to reproduce the problem

```
import qiskit
from qiskit import QuantumCircuit
import qiskit.pulse as pulse
from qiskit.providers.aer.pulse import PulseSystemModel
from qiskit.compiler import assemble,transpile,schedule as scheduler
from qiskit.test.mock import FakeAthens

backend_mock = FakeAthens()
print("Support pulse : "+str(backend_mock.configuration().open_pulse))

q = QuantumCircuit(2,2)
q.h(0)
q.x(1)
q.cx(0,1)
q.measure(0,0)
q.measure(1,1)

q_trans = transpile(q,backend=backend_mock)
schedule = scheduler(q_trans,backend=backend_mock)

sys_mod = PulseSystemModel.from_backend(backend_mock)
qobj = assemble(schedule,backend=backend_mock,system_model = sys_mod)
result = backend_mock.run(qobj).result()
```

### What is the expected behavior?

The code runs without any error.

### Suggested solutions

Real counterparts of those mock backends does not support open pulse. Maybe that's the issue here. However, this could be very handy to have 5qubits mock backends.

**Comment by hhorii** [comment]
I believe this issue is not related to Aer.

@nkanazawa1989 told me that this issue looks similar to https://github.com/Qiskit/qiskit-terra/pull/5410.
This PR was merged in the last Dec but the latest terra does not contains it.

@mtreinish I guess this PR can be merged to [stable-0.16](https://github.com/Qiskit/qiskit-terra/tree/stable/0.16/qiskit/test/mock/backends/athens) branch?

**Comment by UlysseCambier** [comment]
The issue is also very similar to Qiskit/qiskit-aer#1047.

@vvilpas had that solution :

> The problem is that when valencia_model is instantiated, some UchannelLO objects are created, but instead of having scale property as a complex number they end up with a list.
> While we fix this, you can manually handle this doing the following right after valencia_model is created:
> 
> for uclo in valencia_model.u_channel_lo: 
>     uclo[0].scale = complex(*uclo[0].scale)

This solved my issue.
Thanks.
